src/main/java/io/github/mooeypoo/chatmonitor/words/WordManager.java

```diff
@@ -1,71 +1,64 @@
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
+import java.util.Map;
 import java.util.Set;
+import java.util.logging.Logger;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
 
-import org.bukkit.plugin.java.JavaPlugin;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.google.common.collect.ImmutableSetMultimap;
+import com.google.common.collect.SetMultimap;
 
 import io.github.mooeypoo.chatmonitor.configs.ConfigManager;
 import io.github.mooeypoo.chatmonitor.configs.ConfigurationException;
 import io.github.mooeypoo.chatmonitor.configs.GroupConfigInterface;
 
+@NotThreadSafe
 public class WordManager {
-	private HashMap<String, String> wordmap = new HashMap<>();
-	private HashMap<String, ArrayList<String>> mapWordsInCommands = new HashMap<>();
-	private ArrayList<String> allwords = new ArrayList<>();
-	private ArrayList<String> relevantCommands = new ArrayList<>();
-	private JavaPlugin plugin;
+	private final Logger logger;
+	private WordLoader wordLoader;
 	private ConfigManager configManager;
-	
-	public WordManager(JavaPlugin plugin) {
-		this.plugin = plugin;
-		try {
-			this.configManager = new ConfigManager(Paths.get(this.plugin.getDataFolder().getPath()), "ChatMonitor_wordgroup");
-		} catch (ConfigurationException e) {
-			this.plugin.getLogger().warning("Initiation aborted for ChatMonitor. Error in configuration file '" + e.getConfigFileName() + "': " + e.getMessage());
-			return;
-		}
-		this.collectWords();
-	}
-	
+
+	@Nonnull private SetMultimap<String, String> mapWordsInCommands = ImmutableSetMultimap.of();
+	@Nonnull private Map<String, String> wordMap = Map.of();
+
 	// Used for testing
-	public WordManager(Path filepath, String prefix) {
+	public WordManager(Path dataFolder, String prefix, Logger logger) {
+		this.logger = logger;
 		try {
-			this.configManager = new ConfigManager(filepath, prefix);
+			configManager = new ConfigManager(dataFolder, prefix);
+			wordLoader = new WordLoader(configManager, logger);
+			setWords(wordLoader.collectWords());
 		} catch (ConfigurationException e) {
-			this.plugin.getLogger().warning("Initiation aborted for ChatMonitor. Error in configuration file '" + e.getConfigFileName() + "': " + e.getMessage());
-			return;
+			this.logger.warning("Initiation aborted for ChatMonitor. Error in configuration file '" + e.getConfigFileName() + "': " + e.getMessage());
 		}
-		this.collectWords();
 	}
+
 	/**
 	 * Reload the lists and re-process the groups from the config files.
 	 */
 	public void reload() {
-		// Reset lists
-		this.wordmap.clear();
-		this.allwords.clear();
-		this.mapWordsInCommands.clear();
-		this.relevantCommands.clear();
-		
-		// Refresh all configs 
 		try {
+			// Refresh all configs
 			this.configManager.reload();
+			// Redo word collection
+			setWords(wordLoader.collectWords());
```

Comment:
Random style question -- should this (and others in the class, around) not be `this.setWords` since it's a reference to a class method, rather than a parameter inside this method? I know both work, I'm just curious about the "best practices" of when to use `this.` and when not to, as it seems to be different than what I'd usually be used to...?

Reply:
I think being explicit about this made sense 15 years ago when IDE were crap. Nowadays, your IDE will color instance variables and local variables differently, so it should be obvious which is which. And it should provide good enough navigation that even if you need to see the implementation, it should be trivial to get there. In most cases, you shouldn't care anyway where a method is defined when reading the code, it's intent should be clear enough.

In the end, it's a question of personal preference. I don't care much either way. I tend to not use `this` unless it is required  (an instance variable being hidden by a local variable, like in a constructor).